kuzzleio/kuzzle-admin-console
# Home-made end-to-end UI testing with CucumberJS and Chrome Puppeteer

## What does this PR do ?

This PR enables end-to-end testing via Chrome Puppeteer. Tests run in a CucumberJS suite with expect-js and Puppeteer API for assertions.
These tests include interaction tests as well as visual regression tests (performed via blink-diff). Please refer to the `CONTRIBUTING.md` file to learn how they work.
Tests can be run in headless or headful mode.

### How should this be manually tested?

* Remove your `node_modules` directory (unless your local Nodejs version is 8.12.x)
* Stop all your running Kuzzle stack Docker containers
* run `npm test` to run all tests (unit and e2e) or simply look at the Travis build on this PR.
* run `npm run e2e` to run end-to-end tests in headless mode
* run `npm run e2e-local` to run end-to-end tests in headful mode

Be patient, running all the test stack will be long the first time: you'll see ` Waiting for Kuzzle Admin Connsole to be up at http://adminconsole:3000` many many times. Go grab a ☕️ and talk to your peers.
If you really doubt it's working you can

```
cd test/e2e
docker-compose logs -f adminconsole
```

Now that you know what's happening, go grab a ☕️ 

### Other changes

Many components have been enriched with BEM-compliant classes
